atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTopologyValidator.java

```diff
@@ -327,9 +347,9 @@ private Map<CassandraServer, HostIdResult> fetchHostIdsIgnoringSoftFailures(
 
     @VisibleForTesting
     HostIdResult fetchHostIds(CassandraClientPoolingContainer container) {
-        try {
-            return container.<HostIdResult, Exception>runWithPooledResource(
-                    client -> HostIdResult.success(client.get_host_ids()));
+        CassandraServer server = container.getCassandraServer();
+        try (CassandraClient client = cassandraClientFactory.apply(server)) {
```

Comment:
This is the biggest change in the commit - when using the client pool, it contains clients that run setKeyspace to scope the client down to the relevant keyspace. This worked before, because we ran CassandraVerifier methods on the initial pools that were not verified (this was fun, because we actually choose a random host to do this with, so if you have a split brain, you create the keyspace on one, but not the other :/)

However, since we run the validator first, we can't use the client pools. We instead need to use the raw clients without the keyspace set (the same way that CassandraVerifier works). This means that on each refresh, we open fresh clients and then close them. With fast refreshes, this isn't trivial.

An alternative to this is to split the one time setup tasks, create the keyspace on _all_ hosts (even if there's a split brain), then do the validation, then run the second part of setup tasks that do denylisting

Reply:
yeah I'm a bit worried about the cost of this because atlasdb-proxy could spam Cassandra with a few hundred new clients/new sessions every refresh, and some work @inespot and @wi11dey did on one of our internal clusters suggested that cleaning up clients/sessions wasn't free. 

I _think_ wrongly creating the keyspace if there's a split brain isn't the worst thing in the interest of performance, though curious to get your thoughts.